contracts/stellar-multicall/src/tests/test.rs

```diff
@@ -51,198 +59,413 @@ impl TestTarget {
 
     let function_calls = vec![
         &env,
-        FunctionCall {
-            contract: target,
-            function: symbol_short!("failing"),
-            args: Vec::<Val>::new(&env),
-        },
+        construct_function_call!(env, target_id, owner, failing()),
     ];
 
-    client.multicall(&function_calls);
+    let multicall_auth = mock_auth!(owner, client.multicall(&function_calls));
+
+    client
+        .mock_auths(&[multicall_auth])
+        .multicall(&function_calls);
 }
 
 #[test]
 fn multicall_fails_when_target_returns_error() {
-    let (env, client, target, _) = setup();
+    let TestConfig {
+        env,
+        client,
+        target_id,
+        owner,
+    } = setup();
 
     let function_calls = vec![
         &env,
-        FunctionCall {
-            contract: target,
-            function: Symbol::new(&env, "failing_with_error"),
-            args: Vec::<Val>::new(&env),
-        },
+        construct_function_call!(env, target_id, owner, failing_with_error()),
     ];
 
+    let multicall_auth = mock_auth!(owner, client.multicall(&function_calls));
+
     assert_contract_err!(
-        client.try_multicall(&function_calls),
+        client
+            .mock_auths(&[multicall_auth])
+            .try_multicall(&function_calls),
         ContractError::FunctionCallFailed
     );
 }
 
 #[test]
 fn multicall_fails_when_some_calls_returns_error() {
-    let (env, client, target, _) = setup();
+    let TestConfig {
+        env,
+        client,
+        target_id,
+        owner,
+    } = setup();
 
     let function_calls = vec![
         &env,
-        FunctionCall {
-            contract: target.clone(),
-            function: symbol_short!("method"),
-            args: vec![&env, IntoVal::<_, Val>::into_val(&42u32, &env)],
-        },
-        FunctionCall {
-            contract: target.clone(),
-            function: Symbol::new(&env, "failing_with_error"),
-            args: Vec::<Val>::new(&env),
-        },
-        FunctionCall {
-            contract: target,
-            function: symbol_short!("method"),
-            args: vec![&env, IntoVal::<_, Val>::into_val(&0u32, &env)],
-        },
+        construct_function_call!(
+            env,
+            target_id,
+            owner,
+            method(IntoVal::<_, Val>::into_val(&42u32, &env))
+        ),
+        construct_function_call!(env, target_id, owner, failing_with_error()),
+        construct_function_call!(
+            env,
+            target_id,
+            owner,
+            method(IntoVal::<_, Val>::into_val(&0u32, &env))
+        ),
     ];
+
+    let multicall_auth = mock_auth!(owner, client.multicall(&function_calls));
+
     assert_contract_err!(
-        client.try_multicall(&function_calls),
+        client
+            .mock_auths(&[
+                multicall_auth.clone(),
+                multicall_auth.clone(),
+                multicall_auth
+            ])
+            .try_multicall(&function_calls),
```

Comment:
```suggestion
            .mock_all_auths()
            .try_multicall(&function_calls),
```
When testing for a non-auth error, you can mock all auths for simplicity

Reply:
`mock_all_auths()` works for one function call at a time, so in this case with 3 function calls the authorization is incomplete. However, for all tests w/ only 1 function call this works so I applied it there

Reply:
That doesn't quite make sense. What's the reason for it? Take a deeper look, you can talk to others in the team, or ask the Stellar team

Reply:
Talked to the stellar team, and I think the best option is just to leave it as is https://interoplabs.slack.com/archives/C04545F3S59/p1743451741779649?thread_ts=1743450804.674979&cid=C04545F3S59